Sprint 9 solution http api

masmetus opened the pull request.

masmetus committed:
Работа с датами и приоритетами.

masmetus committed:
fix after CheckStyle

masmetus committed:
fix after review:
1. Изменена логика работы с загрузкой задачи
2. Модификатор доступа у мап с данными
3. Тест на новую логику

masmetus committed:
minor fix

masmetus committed:
Задание по 9 спринту.
Сервер + CRUD операции
Адаптеры для даты и статуса
Написаны тесты
Минорные исправления в логике и моделях

z8q commented on src/http/PrioritizedHandler.java:
```diff
@@ -0,0 +1,61 @@
+                .registerTypeAdapter(Duration.class, new DurationAdapter())
+                .setPrettyPrinting()
+                .serializeNulls()
+                .create();
```
чтобы не дублировать в каждом хендлере, можно вынести в BaseHttpHandler,
сделать например protected поле или метод, который создает Gson

z8q commented on src/http/EpicHandler.java:
```diff
@@ -0,0 +1,163 @@
+            sendNotFound(exchange);
+        } catch (Exception e) {
+            sendNotFound(exchange);
+        }
```
можно сложить под один `catch (Exception e)` если обработка одинаковая в обоих случаях

z8q commented on src/http/HistoryHandler.java:
```diff
@@ -0,0 +1,61 @@
+
+    InMemoryTaskManager managers;
+
+    Gson gson;
```
можно поставить private final

название managers предполагает список,
можно назвать в единственном числе

masmetus committed:
Фиксы после ревью:
1) перетащил gson билдер в базовый хендлер
2) объединил блоки catch из-за их дублирования (разные эксепшены, но один ответ)

masmetus committed:
Фиксы после ревью:
1) перетащил gson билдер в базовый хендлер
2) объединил блоки catch из-за их дублирования (разные эксепшены, но один ответ)

masmetus committed:
fix after CheckStyle v2
